mpz-core/src/lpn.rs

```diff
@@ -0,0 +1,180 @@
+//! Implement LPN with local linear code.
+//! More especifically, a local linear code is a random boolean matrix with at most D non-zero values in each row.
+
+use crate::{prp::Prp, Block};
+use rayon::prelude::*;
+/// An LPN encoder.
+///
+/// The `seed` defines a sparse binary matrix `A` with at most `D` non-zero values in each row.
+///
+/// Given a vector `x` and `e`, compute `y = Ax + e`.
+///
+/// `A` - is a binary matrix with `k` columns and `n` rows. The concrete number of `n` is determined by the input length. `A` will be generated on-the-fly.
+///
+/// `x` - is a `F_{2^128}` vector with length `k`.
+///
+/// `e` - is a `F_{2^128}` vector with length `n`.
+///
+/// Note that in the standard LPN problem, `x` is a binary vector, `e` is a sparse binary vector. The way we difined here is a more generic way in term of computing `y`.
```

Comment:
@xiangxiecrypto , would it be possible to add some links to literature which justifies mixing a binary matrix with non-binary vectors? I could not find it with a simple web search.

Reply:
It is a common step in PCG-style COT. You can find the intuition in the [Ferret paper](https://eprint.iacr.org/2020/924.pdf) (page 11) and the references in it.
Note that we can embed a bit into a `Block` to support binary matrix with binary vectors.

> @xiangxiecrypto , would it be possible to add some links to literature which justifies mixing a binary matrix with non-binary vectors? I could not find it with a simple web search.

Reply:
@xiangxiecrypto , thanks, I was worried because I read in
https://eprint.iacr.org/2017/617
"""
How to sample matrices with large dual distance? We suggest to sample a d-sparse matrix M ∈ Fm×k in two steps. First, choose the locations of the non-zero entries of the matrix (e.g., by selecting a random set of d entries per row), and then fill them with random field elements.
"""
since it says  "random field elements" I was assuming that we can't just use "1" as your code does but we must use an element from F_2^128.

Maybe this does not apply to our case, idk. If it's not too laborious to explain, could you shed some light why using "1" instead of a random field element is ok?